doc/usecases/plugins/spec_globbing.md

```diff
@@ -0,0 +1,61 @@
+# Use Case: Specification using Globbing
+
+## Summary
+
+- **Scope:** Configuration
+- **Level:** Developer Goal
+- **Actors:** Application developer
+- **Brief:** Writing generic specification using elektra globbing in key names.
+
+## Scenarios
+
+- **Precondition:** kdb is installed.
+- **Main success scenario:**
+  - Globbing is used in order to make specification generic.
+  - Globbing can be achieved by using wildcard characters.
+  - The used wildcard characters here are `_` and `#`.
+  - In order to write the specification, the kdb tool is used.
+  - To create a key, meta-keys are created immediately as well.
+  - This command `kdb meta-set spec:/service/_/port type string` creates a key `service/_/port` with metadata `type` indicating the `port` 
+  is of type string.
+  - `service/_/port` allows us to create any key name starting with `service/` and ending with `/port`.
+  - A possible example is `service/mailserver/port`.
+  - The usage of this wildcard character `_` is called `globbing`.
```

Comment:
Try something more like this (intentionally incomplete):

```
- The application developer wants to write a specification for their application's configuration.
- The configuration does not have a fixed set of keys.
  Instead it follows a schema, where the same set of keys may occur under different parents.
- To achieve this the application developer uses globbing expressions in their specification, which creates specification keys that apply to many config keys at once.
```

IMO you could also add the part above as a "Base scenario" and start all other scenarios with
```
- based on the base scenario
```

The the "Main scenario" could be an application that uses arrays (`#`) and the alternative scenario could be for named parents (`_`). The error scenario could then state, if the application tries to use arrays and named parents in a conflicting way, Elektra should report an error.